### PR TITLE
Flatnode storage

### PR DESCRIPTION
**Pull request by philipkozeny**
This PR allows the usage of flatnode files (https://nominatim.org/release-docs/3.7.1/admin/Import/#flatnode-files).

As soon as the mount of a volume / bind-mount on `nominatim/flatnode` is available the flatnode config is set and used.

We can't add test cases though, since the flatnode file consumes ~70 GB (even for a small import) and the GitHub Actions are limited to ~30 GB of space. Will do some further tests locally before we merge, but looking for feedback.

Closes https://github.com/mediagis/nominatim-docker/issues/9

### CODE REVIEW

File: 3.7/config.sh [hunk 1]
```diff
@@ -35,3 +35,7 @@ if [ ! -z "$IMPORT_STYLE" ]; then
 else
   sed -i "s|__IMPORT_STYLE__|full|g" ${CONFIG_FILE}
 fi
+
+# if flatnode directory was created by volume / mount, use flatnode files
+
+if [ -d "${PROJECT_DIR}/flatnode" ]; then sed -i 's\NOMINATIM_FLATNODE_FILE=\NOMINATIM_FLATNODE_FILE="/nominatim/flatnode/flatnode.file"\g' ${CONFIG_FILE}; fi
```
Comment (leonardehrenfried) [inline 3.7/config.sh:41]: The documentation is unclear about this: should you keep the file around after import?

**Comment by leonardehrenfried** [comment]
Looks good generally. :+1: 

What happens if someone import with volume + flatnodes and then recreates the container with just the volume? Will it just fail loudly?

**Review by leonardehrenfried — APPROVED** [review]

File: 3.7/config.sh (hunk 1 above)
Comment (philipkozeny) [reply 3.7/config.sh:41]: If you plan to update your data, yes.

**Comment by philipkozeny** [comment]
> Looks good generally. 👍
> 
> What happens if someone import with volume + flatnodes and then recreates the container with just the volume? Will it just fail loudly?

If you don't plan to update your data that's OK, otherwise the update will fail and ask for the flatnode file location.

File: 3.7/config.sh (hunk 1 above)
Comment (philipkozeny) [reply 3.7/config.sh:41]: What I can try: import with flatnode, start container without mount, change setting to disable flatnode and update the data then. Will check and maybe add that change here in addition to catch that case.

@mtmail is there anything that would speak against using a flatnode file only for the initial import?

Comment (leonardehrenfried) [reply 3.7/config.sh:41]: BTW, it's not a requirement for me for this feature to be idiot-proof. If you want to use it, you should know what to do.

Comment (philipkozeny) [reply 3.7/config.sh:41]: True, but maybe it's a viable option I didn't think about before that saves space on your disk.

**Pull request merged**